### PR TITLE
Updating Phoenix library settings

### PR DESCRIPTION
**Pull request by EdwardMoyse**
As part of updating the threejs r167, I ran into some problems, and was advised to update our `tsconfig`. I am now trying to follow this:
https://www.typescriptlang.org/docs/handbook/modules/guides/choosing-compiler-options.html#im-writing-a-library
(have a look at the `tsconfig` changes to see what I did here)

Enabling stricter compilation however caused a VAST number of errors, and I have been slowly working through them. In some cases the fix was obvious, in some I'm really not sure what the original intention was (and so my 'fix' might be wrong). 

I had to use a lot of `object as Mesh` casts, so I hope these work okay (in some cases I check the cast, where I think it might fail).

However I am stuck on this:
```
src/managers/ui-manager/phoenix-menu/phoenix-menu-node.ts(218,11): error TS7053: Element implicitly has an 'any' type because expression of type 'keyof PhoenixMenuConfigs' can't be used to index type 'ConfigLabel | ConfigCheckbox | ConfigSlider | ConfigButton | ConfigColor | ConfigRangeSlider | ConfigSelect'.
  Property 'button' does not exist on type 'ConfigLabel | ConfigCheckbox | ConfigSlider | ConfigButton | ConfigColor | ConfigRangeSlider | ConfigSelect'.
```

Since I'm about to time out on working on this for a bit, let me ask in @9inpachi @sponce @DraTeots if they have any idea?

[Removing question I fixed a while back]

### CODE REVIEW

**Comment by EdwardMoyse** [comment]
Hmm. Maybe there's a bigger problem - if I comment out that line (see latest commits) and fix some issues in `phoenix-ng` then I can compile and start. But `<app-root/>` is empty:
image

**Comment by EdwardMoyse** [comment]
Right, I am hopelessly lost here. I suspect that my efforts to update Phoenix have broken it. But if I look at  https://hepsoftwarefoundation.org/phoenix/#/ I see:
Screenshot 2024-08-26 at 16 23 47

Whilst locally I see:
image

(I added `TEST` just so I could be sure that it was actually picking up the local file). 

So, somehow angular is not running properly? I would expect to see SOME errors/warnings though!

**Comment by 9inpachi** [comment]
@EdwardMoyse try pulling the latest commit and the application should run. I notice issues like the gltf models not being colored and also the tests failing but I think that's related to the other changes that you introduced in the PR. So I will leave you to that.

**Comment by EdwardMoyse** [comment]
Thanks @9inpachi! 

But I see you reverted 'moduleResolution' back to 'node'

As discussed here:

https://github.com/three-types/three-ts-types/issues/1136#issuecomment-2258295707

This is deprecated:

https://www.typescriptlang.org/docs/handbook/modules/reference.html#node10-formerly-known-as-node

And I *think* some of the other changes you reverted came from that (but I'm not positive).

**Comment by EdwardMoyse** [comment]
When I cherry-pick [975521a](https://github.com/HSF/phoenix/pull/675/commits/975521a0f8b1035f89edb7347b44d19bfd9ce948) I get :
```
../../node_modules/@types/three/examples/jsm/loaders/KTX2Loader.d.ts(2,42): error TS2307: Cannot find module 'three/webgpu' or its corresponding type declarations.
  There are types at '/Users/emoyse/LocalDocuments/phoenix/node_modules/@types/three/build/three.webgpu.d.ts', but this result could not be resolved under your current 'moduleResolution' setting. Consider updating to 'node16', 'nodenext', or 'bundler'.

```

I did a `git reset --hard` and then `git clean -fdx` so I'm not sure how you could run but I can't.

**Comment by EdwardMoyse** [comment]
The geometry is now back to the correct colour, as are the EDM objects. However the tests are failing. 

```
  ● Test suite failed to run

    src/tests/helpers/active-variable.test.ts:1:10 - error TS1286: ESM syntax is not allowed in a CommonJS module when 'verbatimModuleSyntax' is enabled.

    1 import { ActiveVariable } from '../../../src/helpers/active-variable.js';
               ~~~~~~~~~~~~~~
```

This looks very painful to fix ... I'm really not sure why we're using CommonJS when ESM is the Typescript default (and which we use internally). :-(

**Comment by EdwardMoyse** [comment]
Hmm. It's interesting that `phoenix-ng` works... I wouldn't have expected this!

**Comment by EdwardMoyse** [comment]
Good grief ... that was an ordeal.

File: packages/phoenix-event-display/src/event-display.ts [hunk 1]
```diff
@@ -357,9 +358,9 @@ export class EventDisplay {
    */
   public async loadGLTFGeometry(
     url: any,
-    name: string | undefined,
-    menuNodeName?: string,
-    scale?: number,
+    name: string,
+    menuNodeName: string = '',
```
[the remark this answers is not in the document]
Comment (EdwardMoyse) [reply packages/phoenix-event-display/src/event-display.ts:362]: Hi @sponce - sorry about that! I *think* the reason was that the codechecks got a lot stricter, and I changed this to remove a warning. But I'm not completely sure. Feel free to revert.

[the remark this answers is not in the document]
Comment (sponce) [reply packages/phoenix-event-display/src/event-display.ts:362]: No problem. I've tired a fix in #725. It work locally

**Pull request merged**